server/src/main/java/org/elasticsearch/repositories/RepositoriesThrottlingStats.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentFragment;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static java.util.stream.Collectors.toMap;
+
+public class RepositoriesThrottlingStats implements Writeable, ToXContentFragment {
+
+    private final Map<String, ThrottlingStats> repositoryToThrottlingStats;
+
+    public RepositoriesThrottlingStats(StreamInput in) throws IOException {
+        repositoryToThrottlingStats = in.readMap(StreamInput::readString, ThrottlingStats::new);
+    }
+
+    public RepositoriesThrottlingStats(Map<String, ThrottlingStats> repositoryThrottlingStats) {
+        repositoryToThrottlingStats = new HashMap<>(repositoryThrottlingStats);
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeMap(repositoryToThrottlingStats, StreamOutput::writeString, (o, v) -> v.writeTo(o));
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.field("repository", repositoryToThrottlingStats);
+        return builder;
+    }
+
+    public Map<String, ThrottlingStats> stats() {
+        return Collections.unmodifiableMap(repositoryToThrottlingStats);
+    }
+
+    public record RepositoryThrottling(String repositoryName, long totalReadThrottledNanos, long totalWriteThrottledNanos) {}
```

Comment:
can't we use this class instead of `ThrottlingStats`? and as I stated above maybe we should name this `RepositoryStats`?

Reply:
Simplified